os_migrate/plugins/module_utils/resource.py

```diff
@@ -0,0 +1,236 @@
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import ansible_collections.os_migrate.os_migrate.plugins.module_utils \
+    as collection_utils
+
+class Resource():
+
+    resource_type = 'UNDEFINED'
+
+    # Took the idea of property lists for serialization from Ryan's
+    # code, will mark the patch co-authored.
+
+    # Serialization
+    params_from_sdk = []
```

Comment:
How often are these param and info names different?  I understand the case where we're doing the name/id mapping, but for the rest of them they are most likely the same always, right?

Reply:
Params and info are usually completely non-overlapping. Perhaps i don't understand where the question is headed though -- is there some potential code amendment that you are thinking of?

Reply:
What I mean here is we have "params_from_sdk" and "sdk_params_from_params".  These items are shown in the Network subclass below to be the same.  Since we're controlling the names used on export,  we know what the names will be on import.  If a name changes in openstacksdk, then we'd probably have a os-migrate version that pins to a specific openstacksdk version.  

There isn't a case now where we are using different keys for these values, and if there was, there would need to be custom code in the serialize and deserialize methods anyway to do the mapping between them.  

tl;dr: we're optimizing for a case that doesn't exist now and is unlikely to exist in the future.  If it did happen to appear, the developer could account for this when writing the subclass.  I think this falls under YAGNI.

Reply:
I see now, yes so for now i reduced the boilerplate into a oneliner `sdk_params_from_params = params_from_sdk`. We are probably going to need something advanced when we get to RouterInterfaces, but it may also be that simple separation won't suffice and we'll have to fully override quite a lot of the Resource base class, so perhaps indeed we could just drop this separation altogether. We can always re-add it if we find the case for it. (The good thing with having it in Resource is that **if** the separation is needed, it allows us to still reuse bits in subclasses rather than reimplementing existing methods and just changing one variable name. But i think i'm starting to agree with the YAGNI point of view.)